PWGHF/D2H/TableProducer/dataCreatorCharmHadPiReduced.cxx

```diff
@@ -824,19 +826,28 @@ struct HfDataCreatorCharmHadPiReduced {
       isHfCandBhadConfigFilled = true;
     }
 
-    // handle normalization by the right number of collisions
-    hfCollisionCounter(collisions.tableSize());
-
+    int zvtxColl {0};
+    int sel8Coll {0};
+    int zvtxAndSel8Coll {0};
+    int allSelColl {0};
     for (const auto& collision : collisions) {
+      float centrality{-1.f};
+      const auto rejectionMask = hfEvSel.getHfCollisionRejectionMask<true, o2::hf_centrality::CentralityEstimator::None>(collision, centrality);
+      if (!TESTBIT(rejectionMask, o2::hf_evsel::EventRejection::Trigger)) sel8Coll++;
+      if (!TESTBIT(rejectionMask, o2::hf_evsel::EventRejection::PositionZ)) zvtxColl++;
+      if (!TESTBIT(rejectionMask, o2::hf_evsel::EventRejection::PositionZ) && !TESTBIT(rejectionMask, o2::hf_evsel::EventRejection::Trigger) ) zvtxAndSel8Coll++;
+      if (rejectionMask == 0) allSelColl++;
       auto thisCollId = collision.globalIndex();
       auto candsCThisColl = candsC.sliceBy(candsDplusPerCollision, thisCollId);
       auto trackIdsThisCollision = trackIndices.sliceBy(trackIndicesPerCollision, thisCollId);
       runDataCreation<false, false, DecayChannel::B0ToDminusPi>(collision, candsCThisColl, trackIdsThisCollision, tracks, tracks, bcs);
     }
+    // handle normalization by the right number of collisions
+    hfCollisionCounter(collisions.tableSize(), zvtxColl, sel8Coll, zvtxAndSel8Coll, allSelColl );
```

Comment:
This block is now duplicated many times. It would be better to factor it out.

Reply:
Working on it! Should I make it into a small function in "utilsEvSelHf.h"?

Reply:
Small function yes, but I would not put it in `utilsEvSelHf.h` as this is specfici to the reduced data model.

Reply:
Ok